### PR TITLE
Fix branch detection of the TestGyms in CI

### PR DESCRIPTION
**Pull request by mironec**

### CODE REVIEW

**Review by simonsarginson — APPROVED** [review]

**Comment by MatthewSandfordImprobable** [comment]
I have very little understanding of what is going on here, I don't think my review is worth much but will approve.

**Review by MatthewSandfordImprobable — APPROVED** [review]

File: ci/setup-build-test-gdk.ps1 [hunk 1]
```diff
@@ -19,20 +19,18 @@ class TestProjectTarget {
...
         $test_gym_version = if (Test-Path -Path $test_gyms_version_path) {[System.IO.File]::ReadAllText($test_gyms_version_path)} else {[string]::Empty}
         if (Test-Path $test_env_override) {
-            $this.test_repo_branch = $test_env_override
+            $this.test_repo_branch = (Get-Item $test_env_override).value
         }
         elseif($testing_repo_heads -Match [Regex]::Escape("refs/heads/$gdk_branch")) {
             $this.test_repo_branch = $gdk_branch
         }
-        elseif(Test-Path $test_gym_version) {
+        elseif($test_gym_version -ne "") {
```
Comment (simonsarginson) [inline ci/setup-build-test-gdk.ps1:33]: very minor, you don't need -ne "" you can just:
`elseif($test_gym_version)`

File: ci/setup-build-test-gdk.ps1 [hunk 2]
```diff
@@ -19,20 +19,18 @@ class TestProjectTarget {
...
-        $this.test_gyms_version_path = $test_gyms_version_path
-        $this.test_env_override = $test_env_override
         
         # Resolve the branch to run against. The order of priority is:
         # envvar > same-name branch as the branch we are currently on > UnrealGDKTestGymVersion.txt > "master".
         $testing_repo_heads = git ls-remote --heads $test_repo_url $gdk_branch
         $test_gym_version = if (Test-Path -Path $test_gyms_version_path) {[System.IO.File]::ReadAllText($test_gyms_version_path)} else {[string]::Empty}
         if (Test-Path $test_env_override) {
-            $this.test_repo_branch = $test_env_override
+            $this.test_repo_branch = (Get-Item $test_env_override).value
```
Comment (simonsarginson) [inline ci/setup-build-test-gdk.ps1:28]: Why is this needed?

Comment (mironec) [reply ci/setup-build-test-gdk.ps1:28]: We are now passing the environment variables by the path where they can be found. Test-Path works on the path as a string, but then we also need a corresponding function to actually retrieve the environment variable's value from that path.

File: ci/setup-build-test-gdk.ps1 (hunk 1 above)
Comment (mironec) [reply ci/setup-build-test-gdk.ps1:33]: Hm, seems like it - however I find this a little bit more explicit, as we are setting it to an empty string above.
I could also reuse the empty string constant.

File: ci/setup-build-test-gdk.ps1 (hunk 2 above)
Comment (simonsarginson) [reply ci/setup-build-test-gdk.ps1:28]: Super minor: reading the docs this seems like the nicest way to do it. https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-itemproperty?view=powershell-7.1#example-4--get-the-value-name-and-data-of-a-registry-entry-in-a-registry-subkey

Comment (mironec) [reply ci/setup-build-test-gdk.ps1:28]: image

Comment (simonsarginson) [reply ci/setup-build-test-gdk.ps1:28]: `Get-ItemProperty -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion -Name "value"`
I think is the format

Comment (mironec) [reply ci/setup-build-test-gdk.ps1:28]: I tried a few ways, seems tailored to registry values or something

Comment (simonsarginson) [reply ci/setup-build-test-gdk.ps1:28]: fair enough
